packages/nextjs/utils/scaffold-eth/networks.ts

```diff
@@ -161,3 +162,20 @@ export function getBlockExplorerTxLink(network: Network, txnHash: string) {
 
 export const getNetworkDetailsByChainId = (chainId: number) =>
   Object.values(NETWORKS).find(val => val.chainId === chainId);
+
+/**
+ * @dev returns the metadata of the chain like id, nativeCurrency etc specified in .env file or logs an error in console if chain mentioned in .env is unsupported
+ */
+export const getConfiguredChainFromENV = () => {
```

Comment:
>  Find a better way for `const chain = NETWORKS[process.env.NEXT_PUBLIC_NETWORK as string] as TChainAttributes` or maybe handle error if chainName is wrong in `env`
 
 To handle this and show useful error in console came up with this function `getConfiguredChainFromENV `  please let me your thoughts, happy to take any other approaches 🙌 
 
 **Benefits** 
-  this functions gets you full metadata of chain configured in env along with autocompletion 
- logs potential error info if you misspelled chain or provide and unsupported chain 

I also tried exploring typed env but since `NEXT_PUBLIC` will be client side so didn't find any good solutions also we loose the advantage of getting whole chain metadata like id, name, ... etc which is helpful and typed

Reply:
Good stuff! I renamed it 5fef117244169b4405cb6ec1b2176e3edc7fcb02 to hide implementation details on the function name

Just a silly nitpick I had, sorry haha